### PR TITLE
Fixing 'payload' part for 17 methods from #41 list.

### PR DESCRIPTION
**Pull request by oshtaier**
Robottelo part will be provided in separate PRs

Progress:

| Method    | Fix payload   | Fix path      | Transform to search |
| --------- | ------------- | ------------- | ------------------- |
| ActivationKey.add_subscriptions  | Done  | Not needed  | Not needed        |
| ActivationKey.content_override  | Done  | Not needed  | Not needed        |
| AbstractDockerContainer.power  | Done  | Already using best one  | Not needed        |
| AbstractDockerContainer.logs  | Done  | Already using best one  | Not needed        |
| ContentViewVersion.promote  | Done  | Not needed  | Not needed        |
| ContentView.publish  | Done  | Not needed  | Not needed        |
| ContentView.set_repository_ids  | Done  | Seems good enough  | Not needed        |
| ContentView.available_puppet_modules  | Not needed  | Not needed  | Not needed        |
| ContentView.add_puppet_module  | Done  | Doesn't make sense to create module like 'Content view puppet modules' as it will make path even worse  | Not needed        |
| ContentView.copy  | Done  | Not needed  | Not needed        |
| ForemanTask.poll  | Seems as internal method that shouldn't be changed  | Doesn't have  | Not needed        |
| Organization.subscriptions  | Not needed  | **Required**, but we don't have proper API paths for all subscription methods  | Doesn't seem as redundant method        |
| Organization.upload_manifest  | Done  | **Required**, but we don't have proper API paths for all subscription methods  | Not needed        |
| Organization.delete_manifest  | Not needed  | **Required**, but we don't have proper API paths for all subscription methods  | Not needed        |
| Organization.refresh_manifest  | Not needed  | **Required**, but we don't have proper API paths for all subscription methods  | Not needed        |
| Organization.sync_plan  | Done  | **Required**, but we don't have proper API paths for all subscription methods  | Not needed        |
| Organization.list_rhproducts  | Should be removed entirely  | Should be removed entirely  | Done        |

### CODE REVIEW

**Comment by oshtaier** [comment]
Ready for review

**Comment by omaciel** [comment]
@elyezer can you take a peek when you get a chance?

File: nailgun/entities.py [hunk 1]
```diff
@@ -793,45 +791,34 @@ def create(self, create_missing=None):
...
-        if stdout is not None:
-            data['stdout'] = stdout
-        if stderr is not None:
-            data['stderr'] = stderr
-        if tail is not None:
-            data['tail'] = tail
         response = client.get(
             self.path(which='logs'),
-            data=data,
+            data=payload if payload else {},
```
Comment (omaciel) [inline nailgun/entities.py]: I wonder if this would work if payload is {}?

File: nailgun/entities.py [hunk 2]
```diff
@@ -2342,13 +2338,11 @@ def upload_manifest(self, path, repository_url=None, synchronous=True):
             completes with any result other than "success".
 
         """
-        data = None
-        if repository_url is not None:
-            data = {u'repository_url': repository_url}
+        path = payload.pop('content')
```
Comment (omaciel) [inline nailgun/entities.py]: Is `path` being used anywhere? Maybe my Github Diff foo is weak...

**Comment by omaciel** [comment]
ACK pending comments

File: nailgun/entities.py (hunk 2 above)
Comment (oshtaier) [inline nailgun/entities.py]: In next line:
```with open(path, 'rb') as manifest:```

Comment (omaciel) [inline nailgun/entities.py]: There goes to show that I'm going blind hehe

**Comment by sthirugn** [comment]
Can you paste the test results please.  The PR of this magnitude will affect a lot of robottelo tests.

**Comment by oshtaier** [comment]
Part of results present here - SatelliteQE/robottelo#2631. Will be updated more tomorrow

File: nailgun/entities.py (hunk 1 above)
Comment (oshtaier) [inline nailgun/entities.py]: Yes, it should:
```
>>> payload = None
>>> payload if payload else {}
{}
```
```
>>> payload = 15
>>> payload if payload else {}
15
```
```
>>> payload = {}
>>> payload if payload else {}
{}
```

**Comment by sthirugn** [comment]
ACK - Great job!

**Comment by elyezer** [comment]
I think that is time to merge this. ACK

**Comment by Ichimonji10** [comment]
Nice job @oshtaier. Thank you.

**Pull request merged**